### PR TITLE
  ci: migrating to GitHub actions

### PR DESCRIPTION
**Pull request by JBAhire**
## Description
Migrating to GitHub actions from CircleCI as per https://github.com/hypertrace/hypertrace/issues/144

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
- https://docs.github.com/en/free-pro-team@latest/actions

### CODE REVIEW

File: .github/workflows/pr-build.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          fetch-depth: 0
+
+      - name: verify-protos
+        run: buf check lint
```
Comment (aaron-steinfeld) [inline .github/workflows/pr-build.yml:62]: breaking change detection is missing (see line 108 from deleted cci config)

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  validate-helm-charts:
```
Comment (aaron-steinfeld) [inline .github/workflows/pr-build.yml:64]: can move to pr-test workflow

File: .github/workflows/pr-test.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          name: integration test reports
+          fail_ci_if_error: true
+          flags: integration
+  
+      - name: copy test reports
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: copyAllReports --output-dir=/tmp/test-reports
+            
+      - name: Archive test reports
```
Comment (aaron-steinfeld) [inline .github/workflows/pr-test.yml:60]: archive + copy not needed

File: .github/workflows/pr-test.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        uses: codecov/codecov-action@v1
+        with:
+          name: unit test reports
+          fail_ci_if_error: true
+          flags: unit
+
+      - name: Integration test
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: jacocoIntegrationTestReport
```
Comment (aaron-steinfeld) [inline .github/workflows/pr-test.yml:46]: Running our int tests actually pulls images too - I guess we can see if we hit rate limiting issues.

File: .github/workflows/pr-build.yml (hunk 1 above)
Comment (JBAhire) [reply .github/workflows/pr-build.yml:62]: I added that now but noticed this yesterday those checks weren't working: https://app.circleci.com/pipelines/github/hypertrace/config-service/28/workflows/2bfa2da2-a524-443d-ad63-a66b121ca06b/jobs/127

Comment (aaron-steinfeld) [reply .github/workflows/pr-build.yml]: They have no output on success - see an example of an api change here that would be flagged: https://github.com/hypertrace/config-service/pull/8/files

**Review by aaron-steinfeld — APPROVED** [review]

File: .github/workflows/pr-build.yml (hunk 1 above)
Comment (JBAhire) [reply .github/workflows/pr-build.yml]: got it

**Pull request merged**